src/api/controllers/db-controller.ts

```diff
@@ -345,20 +359,41 @@ export async function getBlockFromDataStore({
 
 export async function getRosettaBlockTransactionsFromDataStore(
   blockHash: string,
+  indexBlockHash: string,
   db: DataStore
 ): Promise<FoundOrNot<RosettaTransaction[]>> {
+  const blockQuery = await db.getBlock(blockHash);
+  if (!blockQuery.found) {
+    return { found: false };
+  }
+
   const txsQuery = await db.getBlockTxsRows(blockHash);
+  const minerRewards = await db.getMinerRewards({
+    blockHeight: blockQuery.result.block_height,
+  });
 
   if (!txsQuery.found) {
     return { found: false };
   }
-  const transactions = txsQuery.result.map(tx => {
-    const operations = getOperations(tx);
-    return {
+
+  const transactions: RosettaTransaction[] = [];
+
+  for (const tx of txsQuery.result) {
+    const events = await db.getTxEvents({
+      txId: tx.tx_id,
+      indexBlockHash: indexBlockHash,
+      limit: LIMIT_EVENTS_QUERY,
```

Comment:
Has this been tested with the genesis block (height 1)? There's 300k+ stx-events in the genesis block. The postgres query to fetch all of these (`SELECT * from stx_events where block_height = 1`) takes ~10 seconds in our infrastructure, and my guess is that the rosetta endpoint response size would be several megabytes.

Reply:
We did the rosetta-cli test on the entire mainnet network, but starting from index 2 because as you say the response of block 1 was quite heavy due to all the mint events.

Reply:
Should this function have special handling for block 1 to ignore events? This would be a dos vulnerability on the public API.

Reply:
I can surely skip searching for events on block 1, also adding a reasonable limit for the query ?. Would `limit: 5000` be a good upper limit for the events amount in one block ?

Reply:
Yep that sounds okay. The main thing we need is to skip the event querying from the genesis block.

Reply:
ok, done!